src/azure-cli/azure/cli/command_modules/iot/_params.py

```diff
@@ -274,3 +278,19 @@ def load_arguments(self, _):  # pylint: disable=too-many-statements
                    help='Access key for the given IoT Plug and Play repository.')
         c.argument('user_role', options_list=['--role'], arg_type=get_enum_type(UserRole),
                    help='User role of the access key for the given IoT Plug and Play repository.')
+
+    with self.argument_context('iot central app') as c:
+        c.argument('app_name', app_name_type, options_list=['--name', '-n'], id_part='display_name')
```

Comment:
Is `display_name` a valid part of resource id? The comment [here](https://github.com/Azure/msrestazure-for-python/blob/master/msrestazure/tools.py#L106) does not include it.

Reply:
`display_name` is not a valid part of the ARM resource id. Should this be removed? Most of this code is untouched from module `az iotcentral`

Reply:
`id_part` is used to parse the parameter value from ARM resource id when used with --ids "/an/arm/id". If `display_name` is not a valid part of the ARM resource id, it should be removed.

Reply:
fixed